server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java

```diff
@@ -283,17 +283,18 @@ public static void finishSplit(KeyExtent extent,
    * datafilesToDelete are strings because they can be a TabletFile or directory
    */
   public static void addDeleteEntries(KeyExtent extent, Set<String> datafilesToDelete,
-      ServerContext context) {
+      ServerContext context, Ample ample) {
```

Comment:
Is ample not available off of ServerContext? Not sure we need to pass in another param here.

Reply:
Yes but each call to `context.getAmple()` creates a new object.  I added it as a param to prevent creating objects in the loop over `datafilesToDelete`.

Reply:
Can't you just create a local object before that loop?

Reply:
Yeah but the code calling this function is in the loop of a scanner and it will create a new object for every entry.

Reply:
Ah, I see.

Reply:
Perhaps in some future issue, we can make ServerContext hold onto a lazily instantiated Ample object, so we don't have to keep constructing one.